components/style/properties/properties.mako.rs

```diff
@@ -313,6 +313,44 @@ impl PropertyDeclarationBlock {
     }
 }
 
+impl PropertyDeclarationBlock {
+    // Take a declaration block known to contain a single property,
+    // and serialize it
+    pub fn to_css_single_value<W>(&self, dest: &mut W, name: &str)
+        -> fmt::Result where W: fmt::Write {
+        match self.declarations.len() {
+            0 => Err(fmt::Error),
+            1 if self.declarations[0].0.name().to_string() == name => {
+                self.declarations[0].0.to_css(dest)
+            }
+            _ => {
+                // we use this function because a closure won't be `Clone`
+                fn get_declaration(dec: &(PropertyDeclaration, Importance))
+                    -> &PropertyDeclaration {
+                    &dec.0
+                }
+                let shorthand = try!(Shorthand::from_name(name).ok_or(fmt::Error));
+                if self.declarations.iter().all(|decl| decl.0.shorthands().contains(&shorthand)) {
+                    // we must `.collect().iter()` because closures aren't Clone
+                    let success = try!(shorthand.serialize_shorthand_to_buffer(
+                        dest,
+                        self.declarations.iter()
+                                         .map(get_declaration as fn(_) -> _),
```

Comment:
IIRC this wasn't done in other places because of inlining, that's why I suggested the custom iterator.